db/import.go

```diff
@@ -545,20 +545,43 @@ func (i *ImportFilterFunction) EvaluateFunction(ctx context.Context, doc Body, d
 		return false, errors.New("Import filter function returned non-boolean value.")
 	}
 }
-func (db *DatabaseCollectionWithUser) ImportFilterDryRun(ctx context.Context, doc Body, docid string) (bool, error) {
-
-	importFilter := db.importFilter()
-	if docid != "" {
-		docInBucket, err := db.GetDocument(ctx, docid, DocUnmarshalAll)
-		if err == nil {
-			if doc == nil {
-				doc = docInBucket.Body(ctx)
+
+// ImportFilterDryRun Runs a document through the import filter and returns a boolean and error
+func (db *DatabaseCollectionWithUser) ImportFilterDryRun(ctx context.Context, doc Body, importFn string) (bool, error) {
+
+	var shouldImport bool
+	if importFn == "" {
+		importFilter := db.importFilter()
+		if importFilter == nil {
+			return false, base.HTTPErrorf(http.StatusBadRequest, "No import filter specified")
+		}
+		output, err := importFilter.EvaluateFunction(ctx, doc, true)
```

Comment:
This is using the shared pool of ImportFilter which may cause contention with the real import process and the diagnostic API.

Would prefer to be able to run the function in a new instance like we did for the sync function.